src/main/scala/com/datasonnet/Mapper.scala

```diff
@@ -90,32 +91,60 @@ object Mapper {
     (name -> jsonnetLibrary)
   }
 
+  def getParams(name: String, mimeType: String, header: Header): util.Map[String, Object] = {
+    if (header != null) {
```

Comment:
This is an internal method in scala code so there's no reason to let it ever get a null value. Without the null, the entire method can be replaced with:

```
val nameParams = header.getDataFormatParameters().getOrDefault(mimeType, Collections.emptyMap())
val params = new util.HashMap[String, Object](nameParams.getOrDefault(Header.DATAFORMAT_DEFAULT, Collections.emptyMap()))
params.putAll(nameParams.getOrDefault(name, Collections.emptyMap()))
params
```

Reply:
Hmm, no, that is, header cannot be null, but dfparams still can be null.

Reply:
good point.